packages/mobile/capacitor/plugins/secure-filesystem-access/android/src/main/java/org/iota/plugins/securefilesystemaccess/SecureFilesystemAccessPlugin.java

```diff
@@ -0,0 +1,325 @@
+                            beginPath = selected.substring(0, slashIndex);
+                        }
+                        // Android 10+ uses new storage, momentally we define Downloads as default
+                        // appPath = Environment.getExternalStoragePublicDirectory(DIRECTORY_DOWNLOADS);
+                        // TODO add popup on client when the selected dir could not be used and Downloads will.
+                        File appPath;
+                        if (beginPath.equals(DIRECTORY_DOCUMENTS) | beginPath.equals(DIRECTORY_DOWNLOADS)) {
+                            appPath = getDirectory(beginPath);
+                        } else {
+                            // user must select again, stronghold shows error
+                            response.put("selected", "");
+                            call.resolve(response);
+                            return;
+                        }
+
+                        response.put("selected",
+                                appPath.getPath() + File.separator + finalPath + File.separator + fileName);
+                    }
+                }
+            }
+            call.resolve(response);
+        }
+    }
+    private static File getDirectory(String name) {
+        File res;
+        if (DIRECTORY_MUSIC.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_MUSIC);
+        else if (DIRECTORY_PODCASTS.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_PODCASTS);
+        else if (DIRECTORY_RINGTONES.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_RINGTONES);
+        else if (DIRECTORY_ALARMS.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_ALARMS);
+        else if (DIRECTORY_NOTIFICATIONS.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_NOTIFICATIONS);
+        else if (DIRECTORY_PICTURES.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_PICTURES);
+        else if (DIRECTORY_MOVIES.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_MOVIES);
+        else if (DIRECTORY_DOWNLOADS.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_DOWNLOADS);
+        else if (DIRECTORY_DCIM.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_DCIM);
+        else if (DIRECTORY_DOCUMENTS.equals(name))
+            res = Environment.getExternalStoragePublicDirectory(DIRECTORY_DOCUMENTS);
+        else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
+                && DIRECTORY_AUDIOBOOKS.equals(name))
+                res = Environment.getExternalStoragePublicDirectory(DIRECTORY_AUDIOBOOKS);
+        // Environment.getExternalStorageDirectory() Gives us the full path the SDCard
+        else res = Environment.getExternalStorageDirectory();
```

Comment:
it's true for Android 10 only, because from 11+ there are the new Scoped storage. 
I get a path with numbers and Stronghold can do nothing, that's why endup using `requestLegacyExternalStorage=true` on the manifest to force Android 10 to use old storage.

Reply:
I've tested with `getExternalFilesDir()` but it seems that external give us an really deep path that is deleted when the app is uninstalled, and where an average user could not find imo.
We would need `Environment.getExternalStoragePublicDirectory()` even deprecated since we need that Stronghold have permissions to write the backup file on an public accessible folder. Also I could not find a better option that works with Capacitor.